lib/kitchen/directions/bake_chapter_introductions.rb

```diff
@@ -60,6 +61,75 @@ def self.v1_update_selectors(something_with_selectors)
         something_with_selectors.selectors.title_in_introduction_page =
           ".intro-text > [data-type='document-title']"
       end
+
+      def self.v2(book:, chapter_objective_strategy: :default)
+        book.chapters.each do |chapter|
+          chapter_outline_html = ''
+
+          case chapter_objective_strategy
+          when :preexisting_title
+            chapter_objectives = chapter.first('div.chapter-objectives')&.cut
+            if chapter_objectives
+              chapter_objectives.search('div[data-type="title"]')&.trash
+              chapter_objectives.wrap_children('div', class: 'os-note-body')
+              chapter_objectives.prepend(child:
+                <<~HTML
+                  <h3 class="os-title" data-type="title">
+                    <span class="os-title-label">#{I18n.t(:chapter_objectives)}</span>
+                  </h3>
+                HTML
+              )
+              chapter_outline_html = chapter_objectives.cut
+            end
+          when :default
+            outline_items_html = chapter.non_introduction_pages.map do |page|
+              <<~HTML
+                <div class="os-chapter-objective">
+                  <a class="os-chapter-objective" href="##{page.title[:id]}">
+                    <span class="os-number">#{chapter.count_in(:book)}.#{page.count_in(:chapter)}</span>
+                    <span class="os-divider"> </span>
+                    <span data-type="" itemprop="" class="os-text">#{page.title.children[0].text}</span>
+                  </a>
+                </div>
+              HTML
+            end.join('')
```

Comment:
I have a few questions :grin:

Second version is covering all cases. When `chapter_objective_strategy` is set to `default` it means that book has Chapter Outline - new element created from the beggining; when it's set to `preexisting_title` it has Chapter Objectives - element that exists in unbaked file and we modify it during baking, and it could be set to `nil` and then book doesn't have neither Chapter Outline, nor Chapter Objectives. Am I understanding correctly? If so, do we have a plan to remove first version in the future, when we update released recipes?  

I'm asking, because if we want to keep both versions, maybe second one should cover only case with Chapter Objectives? So if book has Chapter Outline recipe should use V1, when it has Chapter Objectives - V2.

Also, if we want to have two versions, maybe they should be split into separate files, to keep the convension. But it 's a small thing :wink:

To sum up, for the first scenario (removing V1 in the future) maybe good solution will be to add comment that V1 is used only in released books, and all new recipes should use V2. 

For the second one (keeping both versions) as I write before, we can split code into two files and remove part for Chapter Outline from V2.

Let me know what do you think, and wheter I understand everything correctly. I might miss something :slightly_smiling_face:

Reply:
I'm sorry, I just noticed that I missed last part of your sentence, that you are going to depreciate the v1 method :facepalm: Sorry again :grin: Will that be done in this PR or in the future? If in the future, I think the comment which I mentioned before might be helpful. Beside that everything looks good :relaxed:

Reply:
For now we're leaving V1 _mostly_ alone because it does some chapter_outline stuff that we stumbled adding to a V2, but I think we could eventually depreciate V1